airflow/example_dags/example_params_ui_tutorial.py

```diff
@@ -0,0 +1,234 @@
+        # The benefit of the Params class definition is that the full scope of JSON schema validation
+        # can be leveraged for form fields and they will be validated before DAG submission.
+        "checked_text": Param(
+            "length-checked-field",
+            type="string",
+            title="Text field with length check",
+            description_html="""This field is required. And you need to provide something between 10 and 30
+            characters. See the
+            <a href='https://json-schema.org/understanding-json-schema/reference/string.html'>
+            JSON schema description (string)</a> in for more details""",
+            minLength=10,
+            maxLength=20,
+            section="JSON Schema validation options",
+        ),
+        "checked_number": Param(
+            100,
+            type="number",
+            title="Number field with value check",
+            description_html="""This field is required. You need to provide any number between 64 and 128.
+            See the <a href='https://json-schema.org/understanding-json-schema/reference/numeric.html'>
+            JSON schema description (numbers)</a> in for more details""",
+            minimum=64,
+            maximum=128,
+            section="JSON Schema validation options",
+        ),
+        # Some further cool stuff as advanced options are also possible
+        # You can have the user entering a dict object as a JSON with validation
+        "object": Param(
+            {"key": "value"},
+            type=["object", "null"],
+            title="JSON entry field",
+            section="Special advanced stuff with form fields",
+        ),
+        # If you want to have static parameters which are always passed and not editable by the user
+        # then you can use the JSON schema option of passing constant values. These parameters
+        # will not be displayed but passed to the DAG
+        "hidden_secret_field": Param("constant value", const="constant value"),
+        # Finally besides the standard provided field generator you can have you own HTML form code
+        # injected - but be careful, you can also mess-up the layout!
+        "color_picker": Param(
+            "#FF8800",
+            type="string",
+            title="Pick a color",
+            description_html="""This is a special HTML widget as custom implementation in the DAG code.
+            It is templated with the following parameter to render proper HTML form fields:
+            <ul>
+                <li><code>{name}</code>: Name of the HTML input field that is expected.</li>
+                <li><code>{value}</code>:
+                    (Default) value that should be displayed when showing/loading the form.</li>
+                <li>Note: If you have elements changing a value, call <code>updateJSONconf()</code>.</li>
+            </ul>
+            Example: <code>&lt;input name='{name}' value='{value}' onchange='updateJSONconf()' /&gt;</code>
```

Comment:
Rather than needing custom form's to use `onchange='updateJSONconf()'` can we change the even to pick up all inputs under the form automatically?

Reply:
I tried to apply this comment but when looking to my own example I realized that a "generic" solution might be hard. I fear this generates more problems than it helps users, because adding the event listeners globally might generate side effects when a custom form has other event listeners and the order is not right.